test/test_metalearning/pyMetaLearn/test_meta_features.py

```diff
@@ -153,464 +152,598 @@ def meta_train_data_transformed(request):
+    categorical = {col: True if X[col].dtype.name == 'category' else False
+                   for col in X.columns}
+    mf = meta_features.calculate_all_metafeatures(
+        X, y, categorical, "2", logger=logging.getLogger('Meta'))
+    assert 52 == len(mf.metafeature_values)
+    expected = {
+        # TODO: sync-up with matthias on landmarking
+        # Check all metafeatures, except landmarking
+        # which gives not approximate results. For example:
+        # 0.7616945996275606 == 0.7617070142768466 ± 7.6e-07
+        # among runs
+        'PCASkewnessFirstPC': 0.41897660337677867,
+        'PCAKurtosisFirstPC': -0.677692541156901,
+        'PCAFractionOfComponentsFor95PercentVariance': 0.2716049382716049,
+        'ClassEntropy': 1.1898338562043977,
+        'SkewnessSTD': 7.540418815675546,
+        'SkewnessMean': 1.47397188548894,
+        'SkewnessMax': 29.916569235579203,
+        'SkewnessMin': -29.916569235579203,
+        'KurtosisSTD': 153.0563504598898,
+        'KurtosisMean': 56.998860939761165,
+        'KurtosisMax': 893.0011148272025,
+        'KurtosisMin': -3.0,
+        'SymbolsSum': 49,
+        'SymbolsSTD': 1.3679553264445183,
+        'SymbolsMean': 1.8846153846153846,
+        'SymbolsMax': 7,
+        'SymbolsMin': 1,
+        'ClassProbabilitySTD': 0.28282850691819206,
+        'ClassProbabilityMean': 0.2,
+        'ClassProbabilityMax': 0.7616926503340757,
+        'ClassProbabilityMin': 0.008908685968819599,
+        'InverseDatasetRatio': 23.63157894736842,
+        'DatasetRatio': 0.042316258351893093,
+        'RatioNominalToNumerical': 5.333333333333333,
+        'RatioNumericalToNominal': 0.1875,
+        'NumberOfCategoricalFeatures': 32,
+        'NumberOfNumericFeatures': 6,
+        'NumberOfMissingValues': 22175.0,
+        'NumberOfFeaturesWithMissingValues': 29.0,
+        'NumberOfInstancesWithMissingValues': 898.0,
+        'NumberOfFeatures': 38.0,
+        'NumberOfClasses': 5.0,
+        'NumberOfInstances': 898.0,
+        'LogInverseDatasetRatio': 3.162583908575814,
+        'LogDatasetRatio': -3.162583908575814,
+        'PercentageOfMissingValues': 0.6498358926268901,
+        'PercentageOfFeaturesWithMissingValues': 0.7631578947368421,
+        'PercentageOfInstancesWithMissingValues': 1.0,
+        'LogNumberOfFeatures': 3.6375861597263857,
+        'LogNumberOfInstances': 6.8001700683022,
+    }
+    for key, value in expected.items():
+        assert mf[key].value == pytest.approx(value)
+
+    # Then do numpy!
+    X, y = fetch_openml(data_id=2, return_X_y=True, as_frame=False)
+    categorical = {i: True if categorical else False
+                   for i, categorical in enumerate(categorical.values())}
+    mf = meta_features.calculate_all_metafeatures(
+        X, y, categorical, "2", logger=logging.getLogger('Meta'))
+    for key, value in expected.items():
+        assert mf[key].value == pytest.approx(value)
+
+    sio = StringIO()
+    mf.dump(sio)
```

Comment:
Hey, what's the reason for this?

Reply:
This was there in the old code, for io. I removed it now. I really do not know why they were there, and because of this, I left them.